### DISCUSSION TITLE
VSCode Extension API showInputBox call producing unexpected behavior

### DISCUSSION THREAD

**Discussion by msorens** [microsoft/vscode-discussions#407] [Extension Development QnA] [answered]
_[Note: I found this more targeted forum after posting on StackOverflow, so copying my question here]_

According to the VSCode extension docs, the [InputBoxOption][1] `ignoreFocusOut` is described thusly:

```
Set to true to keep the input box open when focus moves to 
another part of the editor or to another window. 
This setting is ignored on iPad and is always false.
```

I am not using an iPad, so I can presumably ignore the second sentence.

When I do set this property to true, I see absolutely no behavior change. That is,
as soon as I click in some other window the VSCode input box closes. Am I misinterpreting what this property is supposed to do? Or is this behavior broken?
My typical use looks like this:

```
    state.url = await input.showInputBox({
      ignoreFocusOut: true,
      title: this.title,
      step: 1,
      totalSteps: this.maxSteps,
      value: state.url ?? '',
      prompt: 'Enter remote Git URL',
      validate: this.validateProtocol,
      shouldResume: shouldResume,
    });
```

If there is, in fact, no way to alter that undesirable behavior of the input closing (and thus the command terminating),
is there perhaps a way to resume the command? That is, I have a multi-step dialog of 5 steps. Upon restoring focus to the VSCode window can I continue with, say, step 3, where I left off? I have not seen any indication this is possible but thought I would ask while I am here.

(My system: VSCode/macOS version 1.74.2)

  [1]: https://code.visualstudio.com/api/references/vscode-api#InputBoxOptions

**Comment by r3m0t** [comment, marked as answer]
`input.showInputBox` looks like you copied `MultiStepInput` class from vscode-extension-samples repo. Did you look in its source code if that parameter is being read and passed to VS Code? You might need to open a pr on that repo.

**Reply by isidorn** [reply]
This sounds like a good recommendation for next steps to me.

**Comment by msorens** [comment]
Bingo, that was it; I should have considered that. Thanks, @r3m0t !

**Comment by msorens** [comment]
PR opened: https://github.com/microsoft/vscode-extension-samples/pull/803

(I found a second parameter that was also not passed through.)
